tests/test_api/test_endpoints.py

```diff
@@ -3381,4 +3381,70 @@ def test_create_globus_task(api_user_client):
 
     # Clean up: Delete the inserted Globus task.
     allowances_collection.delete_many(allow_spec)
-    globus.delete_many({"task_id": seeded_record["task_id"]})
\ No newline at end of file
+    globus.delete_many({"task_id": seeded_record["task_id"]})
+
+def test_get_jgi_samples(api_user_client):
+    """Test retrieving all JGI samples via GET /wf_file_staging/jgi_samples endpoint."""
+
+    mdb = get_mongo_db()
+    allowances_collection = mdb.get_collection("_runtime.api.allow")
+    allow_spec = {
+        "username": api_user_client.username,
+        "action": "/wf_file_staging",
+    }
+    allowances_collection.replace_one(allow_spec, allow_spec, upsert=True)
+    jgi_samples = mdb.get_collection("wf_file_staging.jgi_samples")
+
+    # Seed the `jgi_samples` collection with a document.
+    faker = Faker()
+    jgi_records = faker.generate_jgi_samples(3)
```

Comment:
I recommend calling them samples everywhere.

```suggestion
    jgi_sample_records = faker.generate_jgi_samples(3)
```